### PR TITLE
Upgrade React Styleguidist

### PR DESCRIPTION
**Pull request by juanca**
https://www.pivotaltracker.com/story/show/173543280

## PR upkeep checklist

- [x] Change log entry N/A
- [x] Label(s)
- [x] Assignee(s)
- [x] Deployment URL: https://mavenlink.github.io/design-system/upgrade-styleguidist
- [x] (When ready for review) Reviewer(s)

### CODE REVIEW

**Comment by juanca** [comment]
We want to expose our React Styleguidist configuration and re-use it in bigmaven. Do the following:

1. `assetsDir: 'styleguide/assets',` should point at that particular directory without any modification in bigmaven
1. `require:` should point at its assets without any modification from bigmaven:

        require: [
          '@babel/polyfill',
          './styleguide/content/content.css',
        ],
1. `require:` should not include polyfill for IE11.
1. Can we avoid using a `styleguideDir: 'build',` setting?

File: src/components/icon/icon.md [hunk 1]
```diff
@@ -21,17 +21,17 @@ The `Icon` component exposes a few configured properties to fine-tune the access
 
 Whenever the `role="button"` is set, the Icon requires a label/title to express what the button does (e.g. "Remove item").
 
-```jsx
-const icon = require('../../svgs/icon-clear-small.svg').default;
+```js
+import icon from '../../svgs/icon-clear-small.svg';
```
Comment (Bloodyaugust) [inline src/components/icon/icon.md:25]: This is excellent! 👍

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ A set of [React](https://reactjs.org/) components created by, and for, [Mavenlin
 
 - Use in your project
 
-  ```jsx
+  ```js
```
Comment (Bloodyaugust) [inline README.md:57]: Why are we swapping all of our `jsx` for `js`?

File: README.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ A set of [React](https://reactjs.org/) components created by, and for, [Mavenlin
 - Setup [CSS modules](https://github.com/css-modules/css-modules). One way of doing that is with [`style-loader` for Webpack](https://github.com/webpack-contrib/style-loader) and [`css-loader` for Webpack](https://github.com/webpack-contrib/css-loader).
 - Setup [SVG processing](https://svgontheweb.com/#spriting). One way of doing that is with [`svg-sprite-loader` for Webpack](https://github.com/kisenka/svg-sprite-loader).
 
-  ```js
+  ```js static
```
Comment (Bloodyaugust) [inline README.md:21]: What is the `static` keyword doing?

File: package.json [hunk 4]
```diff
@@ -56,7 +55,7 @@
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
     "react-ga": "^2.5.6",
-    "react-styleguidist": "^8.0.6",
+    "react-styleguidist": "10",
```
Comment (Bloodyaugust) [inline package.json:58]: 🎉

File: src/components/section/section.md [hunk 5]
```diff
@@ -2,15 +2,19 @@
...
   title="I am a section"
   description="Test description"
 />
 ```
 
 `Section`s can contain child elements, like any other element:
-```
+
+```js
+import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
```
Comment (Bloodyaugust) [inline src/components/section/section.md:15]: Do we not get every component globally anymore?

File: src/components/tag/tag.md [hunk 6]
```diff
@@ -20,15 +20,15 @@ This component's accessibility was built using the [WAI ARIA Examples](https://w
 
 ### Basic examples
 
-```
+```js padded
```
Comment (Bloodyaugust) [inline src/components/tag/tag.md:23]: How about `padded`? 😅

File: README.md (hunk 2 above)
Comment (juanca) [reply README.md:57]: Yeah! I want to keep it as simple and consistent with React Styleguidist docs.

File: README.md (hunk 3 above)
Comment (juanca) [reply README.md:21]: It's part of the React Styleguidist docs: all `js` is rendered into the page (e.g. code examples) but adding `js static` skips the rendering. It is useful for showing code without an actual rendering -- this has confused engineers when reading a guide on how to document a new component. I figured I should keep using `static` to spread the knowledge.

File: src/components/section/section.md (hunk 5 above)
Comment (juanca) [reply src/components/section/section.md:15]: Correct!

File: src/components/tag/tag.md (hunk 6 above)
Comment (juanca) [reply src/components/tag/tag.md]: This isn't doing anything. I don't get it. It is suppose to add padding between examples but I don't know how to get it to work properly.

**Pull request merged**